### ISSUE TITLE
v6.0.106: Build failed after hashes removal : Unable to locate repository with working directory that contains directory '/<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate'.

### ISSUE THREAD

**Issue by mirespace** [dotnet/source-build#2898]
Hi Team!

I'm building v6.0.106 from source-build on Ubuntu (22.xx). After the removal of the hashes  on src/<repos> folder, I'm facing the following error:

```
       7>_GetAllRestoreProjectPathItems:
           Determining projects to restore...
         Restore:
           Restoring packages for /<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate/SourceBuildIntermediate.proj...
           Installed Microsoft.Build.Tasks.Git 1.1.0-beta-21480-02 from /<<PKGBUILDDIR>>/packages/previously-source-built/ with content hash N7/f6hI5YNbbh5d+AkKfPxy/zqubn7hHIE6KOX6isZWpyhi3TlnpZlnFvRJQlIZtM7fOgdMvfcxfSeynHCvzNA==.
           Installed Microsoft.SourceLink.AzureRepos.Git 1.1.0-beta-21480-02 from /<<PKGBUILDDIR>>/packages/previously-source-built/ with content hash wJ9xcLzVheLbysMea3zdUnKajNuHpUKIUPMJ9fJLJWEfhRza9gvciQa7IJg9MXlskKMVLqw9BBBLSH1vECUhvQ==.
           Installed Microsoft.SourceLink.Common 1.1.0-beta-21480-02 from /<<PKGBUILDDIR>>/packages/previously-source-built/ with content hash lKAFyklMHz+0oMXPGlVANw9xAr2yJA7f0/I1bGW9Oux8KP4Z1FIGEIS+cMUgQIIWoLW+N6uD0BIeih9pTI6pjg==.
           Installed Microsoft.SourceLink.GitHub 1.1.0-beta-21480-02 from /<<PKGBUILDDIR>>/packages/previously-source-built/ with content hash m4wYjZWN4DNIIZdUBVOvMC6ZSX8SnAxljWYtlzriistbBKpqCLKdJ+Y0LtbIG1oxCc+KdBxIpKgacf6GY/gZZQ==.
           Committing restore...
           Generating MSBuild file /<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/SourceBuildIntermediate/SourceBuildIntermediate.proj.nuget.g.props.
           Generating MSBuild file /<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/SourceBuildIntermediate/SourceBuildIntermediate.proj.nuget.g.targets.
           Writing assets file to disk. Path: /<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/SourceBuildIntermediate/project.assets.json
           Restored /<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate/SourceBuildIntermediate.proj (in 117 ms).
           
           NuGet Config files used:
               /<<PKGBUILDDIR>>/src/source-build-reference-packages/NuGet.config
               /tmp/tmp.33WGunIgLa/.nuget/NuGet/NuGet.Config
           
           Feeds used:
               /<<PKGBUILDDIR>>/artifacts/obj/x64/Release/blob-feed/packages/
               /<<PKGBUILDDIR>>/packages/reference/
               /<<PKGBUILDDIR>>/packages/previously-source-built/
               /<<PKGBUILDDIR>>/packages/prebuilt/
           
           Installed:
               4 package(s) to /<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate/SourceBuildIntermediate.proj
       7>Done Building Project "/<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate/SourceBuildIntermediate.proj" (Restore target(s)).
       6>GetLicenseFileForIntermediateNupkgPack:
           Copying file from "/<<PKGBUILDDIR>>/src/source-build-reference-packages/LICENSE.txt" to "/<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate/LICENSE.txt".
         PackSourceBuildIntermediateNupkgs:
           Building intermediate nupkg...
       6>Project "/<<PKGBUILDDIR>>/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/6.0.0-beta.22161.1/tools/SourceBuild/AfterSourceBuild.proj" (6) is building "/<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate/SourceBuildIntermediate.proj" (7:7) on node 1 (Pack target(s)).
     7:7>Project "/<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate/SourceBuildIntermediate.proj" (7:7) is building "/<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate/SourceBuildIntermediate.proj" (7:9) on node 1 (_GetFrameworkAssemblyReferences target(s)).
       7>PrepareForBuild:
           Creating directory "/<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/bin/SourceBuildIntermediate/Release/netstandard2.0/".
       7>Done Building Project "/<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate/SourceBuildIntermediate.proj" (_GetFrameworkAssemblyReferences target(s)).
       7>/<<PKGBUILDDIR>>/packages/restored/microsoft.build.tasks.git/1.1.0-beta-21480-02/build/Microsoft.Build.Tasks.Git.targets(25,5): error : Unable to locate repository with working directory that contains directory '/<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate'. [/<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate/SourceBuildIntermediate.proj]
  ##vso[task.logissue type=error;sourcepath=/<<PKGBUILDDIR>>/packages/restored/microsoft.build.tasks.git/1.1.0-beta-21480-02/build/Microsoft.Build.Tasks.Git.targets;linenumber=25;columnnumber=5;code=;](NETCORE_ENGINEERING_TELEMETRY=AfterSourceBuild) Unable to locate repository with working directory that contains directory '/<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate'.
       7>Done Building Project "/<<PKGBUILDDIR>>/src/source-build-reference-packages/artifacts/obj/ArcadeGeneratedProjects/SourceBuildIntermediate/SourceBuildIntermediate.proj" (Pack target(s)) -- FAILED.
       6>Done Building Project "/<<PKGBUILDDIR>>/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/6.0.0-beta.22161.1/tools/SourceBuild/AfterSourceBuild.proj" (default targets) -- FAILED.
       1>Done Building Project "/<<PKGBUILDDIR>>/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/6.0.0-beta.22161.1/tools/Build.proj" (default targets) -- FAILED.
  Deferred Messages
```

I'm building using previous artifacts for the SDK (checking the building also in parallel  without the `--with-sdk`):
```
VERBOSE=1 ./build.sh \
	--with-sdk $(CURDIR)/previously-built-dotnet \
    	-- \
    	/v:n \
    	/p:SkipPortableRuntimeBuild=true \
    	/p:LogVerbosity=n \
    	/p:MinimalConsoleLogOutput=false \
    	/p:ContinueOnPrebuiltBaselineError=true \
```

I think it is due to the hash removal (https://github.com/dotnet/installer/pull/13726  https://github.com/dotnet/installer/pull/13745 ), as I was implementing it on my own and I faced exactly the same.

Could you please take a look at it? If I missed providing some info, please let me know.

Thanks in advance!

Miriam

**Comment by MichaelSimons** [comment]
@mirespace - I am not able to reproduce this locally with 6.0.106 using 6.0.105 packages and sdk.  Can you specify the specific steps you took before running the build.sh command?  How you are specifying/setting up the previous source-built artifacts?  How did you produce the previously-build-dotnet?  How did you produce the tarball?

**Comment by mirespace** [comment]
Hi @MichaelSimons! I was checking all that and found that this error is caused by cleaning all .git* folders inside src/repo.hash folders after tarball creation (`./build.sh /p:ArcadeBuildTarball=true /p:TarballDir="$(readlink -f ../"${unmodified_tarball_name}")" /p:CleanWhileBuilding=true`) ... because when I was retrieving that information for you (the one in the previous comment) I had to replicate the behaviour with the previous tag v6.0.105 (because v6.0.106 was not working) and I encountered the same error (and I was able to build it before, so something different had to be the reason then).

I couldn't test it with the current tag (v6.0.106) because of https://github.com/dotnet/installer/issues/13998., and althought I'm thinking is not because the hashes changes I would like to  check the process with that tag before closes the issue.

I cleaned up the .git* folders because I was asked to do so. It's very curious that something inside the internal git folders for the git info is used in the source build.... could I find more information about this somewhere?

Sorry for the noise.

**Comment by omajid** [comment]
>  It's very curious that something inside the internal git folders for the git info is used in the source build.... could I find more information about this somewhere?

One reason to keep that information around is for [source-link](https://devblogs.microsoft.com/dotnet/producing-packages-with-source-link/). When users are using dotnet and run into exceptions, the desired behaviour is that the stack trace will include complete URLs to the git repo with the file and line that's causing the exception. It's not too different from debian's `dbg` packages that provide users access to the complete information about the file/line that's causing an error.

Some folks have reported success trimming down the contents of `.git` (https://github.com/dotnet/source-build/issues/2780) but we can't fully remove it without breaking a few things.

**Comment by MichaelSimons** [comment]
[Triage] closing per @mirespace's previous comment.
